src/braket/circuits/circuit.py

```diff
@@ -900,6 +978,16 @@ def as_unitary(self) -> np.ndarray:
 
         return calculate_unitary(qubit_count, self.instructions)
 
+    @property
+    def qubits_frozen(self) -> bool:
+        """bool: Whether the circuit's qubits are frozen, that is, cannot be remapped.
+
+        This may happen because the circuit contains compiler directives preventing compilation
+        of a part of the circuit, which consequently means that none of the other qubits can be
+        rewired either for the program to still make sense.
+        """
+        return self._has_compiler_directives
```

Comment:
Does every compiler directive imply frozen qubits?

Reply:
No, and that's why `_has_compiler_directives` is private; the actual check happening under the hood can change in the future.